components/butterfly/src/server/inbound.rs

```diff
@@ -17,16 +17,32 @@
 //! This module handles all the inbound SWIM messages.
 
 use std::net::{SocketAddr, UdpSocket};
-use std::sync::atomic::Ordering;
 use std::thread;
 use std::time::Duration;
 
+use prometheus::{CounterVec, GaugeVec};
+
 use super::AckSender;
 use member::Health;
 use server::{outbound, Server};
 use swim::{Ack, Ping, PingReq, Swim, SwimKind};
 use trace::TraceKind;
 
+lazy_static! {
+    static ref SWIM_MESSAGES_RECEIVED: CounterVec = register_counter_vec!(
+        "hab_butterfly_swim_messages_received_total",
+        "Total number of SWIM messages received",
+        &["type"]
+    )
+    .unwrap();
+    static ref SWIM_BYTES_RECEIVED: GaugeVec = register_gauge_vec!(
+        "hab_butterfly_swim_received_bytes",
```

Comment:
This metric is probably served best if it is a counter.  You can get `rate(hab_butterfly_swim_received_bytes)` in Prometheus server

Reply:
On second thought it looks like you want to track the size for each message, if so you may want a summary as you can get statistical analysis of the sizes.

Reply:
The rust client library doesn't support summary as a metric type.  I used [this example](https://github.com/pingcap/rust-prometheus/blob/master/examples/example_hyper.rs) as my inspiration for some of these metrics, and they use a gauge for tracking bytes.  That jives with [the Prometheus docs for metric types](https://prometheus.io/docs/concepts/metric_types/) which says:
> A gauge is a metric that represents a single numerical value that can arbitrarily go up and down.